tests/unit_tests/sql/transpile_to_dialect_test.py

```diff
@@ -0,0 +1,271 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Tests for transpile_to_dialect function in superset/sql/parse.py
+"""
+
+import pytest
+
+from superset.sql.parse import transpile_to_dialect
+
+
+def test_postgres_single_quote_escaping():
```

Comment:
Let's simplify these tests by (1) using parameters and (2) checking the whole value of `result`. Something like:

```python
@pytest.mark.parametrize(
    "sql, dialect, expected",
    [
        ("name = 'O''Hara'", "postgresql", "name = 'O''Hara'"),
        ("name = 'O''Hara'", "databricks", "name = 'O\'Hara'"),
        ...
    ],
)
def test_single_quote_escaping(sql: str, dialect: str, expected: str) -> None:
    assert transpile_to_dialect(sql, dialect) == expected
```